guest/uml-init.sh

```diff
@@ -22,26 +22,19 @@ exit_poweroff() {
 	if [[ -n "${UML_RET:-}" ]]; then
 		echo "$1" > "${UML_RET}"
 	fi
-	exec poweroff -f
+	exec /usr/sbin/poweroff -f
 }
 
 if [[ -z "${UML_UID:-}" ]]; then
 	echo "ERROR: This must be launched by uml-run.sh" >&2
 	exit_poweroff 1
 fi
 
-if [[ -z "${INVOCATION_ID:-}" ]]; then
-	echo "ERROR: This must be launched by systemd" >&2
-	exit_poweroff 1
-fi
-
-UML_EXEC="$(< /proc/cmdline)"
-UML_EXEC="${UML_EXEC#* --}"
-
-if [[ -z "${UML_EXEC}" ]]; then
-	echo "ERROR: Missing command" >&2
-	exit_poweroff 1
-fi
+# Setup usually done by an actual init system (we use bash directly because
+# we just want to run a single commandline, and this avoids adding a
+# dependency on a given init system on the host filesystem)
```

Comment:
A lot of mount points are missing, see eb32b5073fd451d7f52abcc2f9565855028861ba

They may not be required when running a kselftest, but they are useful for debugging or launching more complex binaries.

Reply:
Yep, I was too focused on running e.g. `check-linux.sh`.